README.md

```diff
@@ -2,5 +2,73 @@
 
 # EGI trust anchors container
 
-This is a [CENTOS 7 utility container](https://hub.docker.com/r/indigoiam/egi-trustanchors) that contains fetch-crl and other utilities
-to provide up-to-date trust anchors to relying applications.
+This is [container](https://hub.docker.com/r/indigoiam/egi-trustanchors) contains fetch-crl and other utilities
+to provide up-to-date trust anchors to relying applications, like Nginx, and to VOMS Attribute Authority service. When run,
+the container updates 2 components:
+
+- The system trust anchors, known as CA bundle, typically found in /etc/pki on an EL-based system
+- The EGI trust anchors maintained by fetch-crl and typically found in /etc/grid-security/certificates
+
+These 2 components are exported as 2 separate volumes mapped when the container is launched and used as volumes read by other
+containers needing these informations.
+
+The container is currently based on AlmaLinux 9.
+
+
+## Configuration
+
+The main configuration needed is the declaration of the 2 volumes. To avoid that the volumes are destroyed when the container is
+deleted, and thus become unavailable to the containers using them, it is recommended to create the volumes before launching
+the containers and to declare the volumes as `external` in the Docker/Podman configuration. It is typically done with (volume names are
+free but must be consistent across the configuration):
+
+```bash
+# podman command can be used if using Podman instead of Docker
+docker volume create cabundle
+docker volume create trustanchors
+```
+
+It is also necessary to define the following environment variable when launching the container so that fetch-crl is run as startup:
+
+```
+FORCE_TRUST_ANCHORS_UPDATE=1
+```
+
+
+## Compose file
+
+Wheter using Docker or Podman, it is recommended to use a compose file to declare the container and how to run it. A typical
+compose section for running this container would be:
+
+```yaml
+services:
+    trustanchors:
+        image: docker.io/indigoiam/egi-trustanchors:main
+        container_name: egi-trustanchors
+        environment:
+            - FORCE_TRUST_ANCHORS_UPDATE=1
+        volumes:
+                - trustanchors:/etc/grid-security/certificates
+                - cabundle:/etc/pki
+
+volumes:
+    cabundle:
+        external: true
+    trustanchors:
+        external: true
+
+
+```
+
+## Cron job
+
+The container exits after updating the trust anchors so it is necessary to add a cron job that periodically restarts the container
+(typically once a day). If the container has been created at startup, for example by executing the compose file, it is enough to
+restart the container with an entry like:
+
+```
+# podman command can be used if using Podman instead of Docker
+0 3 * * * root (date --iso-8601=seconds --utc; (/usr/bin/docker restart egi-trustanchors)) >> /var/log/egi-trustanchors-update.log 2>&1
```

Comment:
I'm not sure what restart does, but it may play badly with this [line](https://github.com/indigo-iam/egi-trust-anchors-container/blob/cee1a17e75306fc35ca4357c0a0df736af62ffed/update-trust-anchors.sh#L26), note the `>>`. Please check that the `tls-ca-bundle-all.pem` file doesn't keep increasing. On the other hand I wonder why there is an append there.

Reply:
`restart` works if the container has already been started once (at startup for example) and is still present but exited... I have to double check because in Podman at least, after some time it doesn't restart anymore because of a problem with the network... I'll try to update this after I clarified the issue I was yesterday.

Reply:
@giacomini about your concern regarding `tls-ca-bundle-all.pem` size continuously increasing, I don't understand why it would happen. `update_ca_trust` is not using the existing version of `tls-ca-bundle-all.pem` as an input but the source files that were used to build it. So you can run it as many times as you want without an impact on the size (as long as the source files are not modified).

Reply:
The script uses `>>` instead of `>`, which would append to a possibly existing `tls-ca-bundle-all.pem`. I don't know the reason why it's written that way, but I just want to be sure that it doesn't cause trouble in case of a container restart. Personally, I would use `>` to recreate the bundle from scratch every time the script is run.

Reply:
You're perfectly right! I didn't check carefully the code. I think we need to change this (as you, I think it is a non sense) but to avoid any breaking change, I'd propose to add an environment variable that will allow to chose between appending or replacing `tls-ca-bundle-all.pem` and keep the current behavior as the default. I'll try to open a PR implementing this tomorrow.

Reply:
I wonder how the current behaviour (appending) may be useful, though.

Reply:
I thought exactly the same after writing my comment. I may implement the opposite: change the default but keep a way to restore the old behavior in case there is a good reason for it that we missed. And once we validated there is none, just remove it.